### PR TITLE
JAX-RS Client Support

### PR DESCRIPTION
**Pull request by trajano**
This pull request adds JAX-RS Client Support.  In addition this will also remove the dependency on SLF4J.

### CODE REVIEW

File: jest-rs/README.md [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+```
+
+Add Jest as a dependency to your project.
+
+
+``` xml
+<dependency>
+  <groupId>io.searchbox</groupId>
+  <artifactId>jest</artifactId>
+  <version>0.1.6</version>
```
Comment (VlasShatokhin) [inline jest-rs/README.md]: shouldn't be 0.1.7?

Comment (kramer) [inline jest-rs/README.md]: Should be 1.0.0-SNAPSHOT but well the whole thing has bigger problems than that.

**Comment by kramer** [comment]
Thank you for your contribution; while I believe the PR was created with good intentions it has severe issues that prevents merging as it currently stands:

* Before anything: the changes should obviously **not** break the build.
* Slf4j and logging related changes should not be in this PR.
* (Sub)Readme should not just be a blind clone instead it should it should have meaningful description of the jax-rs package, point out any usage and dependency differences and refer to the main readme when the details are similar.
* Why is there a `lib` folder with binaries committed in it?
* `JestClientFactory` ignores a huge number of config options.

File: jest-rs/src/main/java/io/searchbox/rs/client/http/JestRsClient.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+/**
+ * @author Dogukan Sonmez
+ * @author cihat keser
+ */
+public class JestRsClient extends AbstractJestClient implements JestClient {
+
+    private final static Logger log = Logger.getLogger(JestRsClient.class.getName());
+    // private final static HttpEntity DEFAULT_OK_RESPONSE = EntityBuilder.create().setText("{\"ok\" : true, \"found\" : true}").build();
+    // private final static HttpEntity DEFAULT_NOK_RESPONSE = EntityBuilder.create().setText("{\"ok\" : false, \"found\" : false}").build();
```
Comment (kramer) [inline jest-rs/src/main/java/io/searchbox/rs/client/http/JestRsClient.java]: By skipping this logic you simply created the inconsistency where the rs package has different behaviour for same use cases (actions), as you can guess that is a huge no-no.

In any case there is no point in having commented out code committed...

File: jest-rs/src/main/java/io/searchbox/rs/client/http/JestRsClient.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            } catch (IOException e) {
+                failed(e);
+            }
+            if (jestResult != null) resultHandler.completed(jestResult);
+        }
+
+        @Override
+        public void failed(final Throwable ex) {
+            log.log(Level.SEVERE, "Exception occurred during async execution.", ex);
+            // JEST-COMMON should really accept throwable not just exception
```
Comment (kramer) [inline jest-rs/src/main/java/io/searchbox/rs/client/http/JestRsClient.java]: If you would like to discuss it please open an issue, having this comment within the code serves no useful purpose.

**Comment by trajano** [comment]
NP I am not sure why it breaks Travis though and when I try to look at the logs it freezes my browser I guess it is just a volume.

I have to check why the lib was there, it occurs when I run the test I thought I skipped it already but I guess it came back will have to remove it.

**Comment by trajano** [comment]
I created it as a separate one like the "droid" version I wasn't intending to make a complete change for it because Apache HTTP Client has a few nice features that can fine tune the connection, but since I was on a JAVA EE7 environment, I wanted to remove the dependencies.

File: jest-rs/src/main/java/io/searchbox/rs/client/http/JestRsClient.java (hunk 2 above)
Comment (trajano) [inline jest-rs/src/main/java/io/searchbox/rs/client/http/JestRsClient.java]: Agreed, I had it commented out because I was trying to figure out what it was used for.  But I should be removing it anyway since the unit tests still pass without it.

File: jest-rs/src/main/java/io/searchbox/rs/client/http/JestRsClient.java (hunk 3 above)
Comment (trajano) [inline jest-rs/src/main/java/io/searchbox/rs/client/http/JestRsClient.java]: The comment was there as a note for myself to change the code when the jest-common API gets changed.  I do not want to introduce an API interface change on jest-common if I can avoid it.

The SLF4J changes from what I can tell do not change the API just the implementation on how logging works.

File: jest-rs/src/main/java/io/searchbox/rs/client/http/JestRsClient.java (hunk 2 above)
Comment (kramer) [inline jest-rs/src/main/java/io/searchbox/rs/client/http/JestRsClient.java]: Checking it again, you're right; none of the tests fail if that logic is removed. I believe it was put in place because parsing logic in action classes did not null check before but they do now. This can be removed safely, I'll also remove it on other clients too.

Comment (trajano) [inline jest-rs/src/main/java/io/searchbox/rs/client/http/JestRsClient.java]: Speaking of which I actually figure it out and put it in another commit.

Comment (kramer) [inline jest-rs/src/main/java/io/searchbox/rs/client/http/JestRsClient.java]: Let's go with removing it :)

File: jest-rs/pom.xml [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            <artifactId>lucene-codecs</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>io.searchbox</groupId>
+            <artifactId>jest-common</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.glassfish.main.extras</groupId>
```
Comment (kramer) [inline jest-rs/pom.xml]: I couldn't find where this is used?

Comment (trajano) [inline jest-rs/pom.xml]: you're talking about glassfish?  That's to provide the implementation for the JAX-RS.  It is only for testing.  The JEE7 container will provide the actual implementation.

**Comment by trajano** [comment]
It seems that the default HEAD responses were removed in master.  I should try to remove it on my branch soon

**Comment by trajano** [comment]
The integration tests I would like to move to a different name space, but it had too much access to `protected` methods in common so I kept it there in the end.

**Comment by kramer** [comment]
Closing without merging as there are outstanding issues as pointed out earlier and no updates were provided in a long time. Please feel free to re-open when you get to resolve said issues.

**Comment by trajano** [comment]
Yup I agree. I took the concept you had though and implemented a pure JAX-RS  client implementation with no dependencies. So it is not as complete as yours but does what I require and have no external dependencies. Thanks.
